javalib/src/main/scala/java/lang/process/GenericProcess.scala

```diff
@@ -1,7 +1,163 @@
 package java.lang.process
 
 import scala.scalanative.unsafe._
+import scala.scalanative.meta.LinktimeInfo
+
+import java.time.{Duration, Instant}
+import java.{util => ju}
+import ju.{Optional, Arrays}
+import ju.concurrent.{CompletableFuture, ConcurrentHashMap, TimeUnit}
+
+abstract class GenericProcess() extends java.lang.Process {
+  private lazy val handle = new GenericProcess.Handle(this)
 
-abstract class GenericProcess extends java.lang.Process {
   private[lang] def checkResult(): CInt
+  private[process] def processInfo: GenericProcess.Info
+
+  override def toHandle(): ProcessHandle = handle
+
+  override def onExit(): CompletableFuture[Process] =
+    toHandle().onExit().thenApply(_ => this)
+
+  override def info(): ProcessHandle.Info = processInfo
+  override def pid(): scala.Long = processInfo.pid
+}
+
+private object GenericProcess {
+  // Represents ProcessHandle for process started by Scala Native runtime
+  // Cannot be used with processes started by other programs
+  class Handle(val process: GenericProcess) extends ProcessHandle {
+    override def parent(): Optional[ProcessHandle] = Optional.empty()
```

Comment:
IMO `Optional.empty()` for this evolution.  I could create
an Issue to suggest an improvement for a future evolution
in more detail.

Having the parent PID (ppid) is useful at a number of levels to
detect reused pids.  More in the `compareTo()` section below.

On unixLike, it is easy enough to get the ppid using posixlib unistd `getppid()`.  From a hasty search, it appears that getting a ppid on
Windows is no easy task.

Since the code calling `UnixProcess` or `WindowsProcess` knows, or
can know, its own pid as well as the child pid, it is creating, it could
pass its own pid as a new argument and avoid fussing with
system calls.  A thought.

Reply:
When having access to parent or children we'd need to move an implementation of `destroy`, `isAlive` and others methods we'd need to move and adapt implementation for these methods from Process to ProcessHandle (or duplicate it). Let's leave this problem for another day